apps/api-journeys/src/app/modules/event/button/button.resolver.ts

```diff
@@ -59,19 +76,31 @@ export class ChatOpenEventResolver {
       input.stepId
     )
 
-    if (visitor.messagePlatform == null) {
-      void this.visitorService.update(visitor.id, {
-        messagePlatform: input.value ?? undefined
+    const lastChatStartedAt = new Date().toISOString()
+    const promiseArray = [
+      this.eventService.save({
+        ...input,
+        __typename: 'ChatOpenEvent',
+        visitorId: visitor.id,
+        createdAt: new Date().toISOString(),
+        journeyId
+      }),
+      this.visitorService.update(visitor.id, {
+        lastChatStartedAt,
+        lastChatPlatform: input.value ?? undefined,
+        lastEventAt: lastChatStartedAt
       })
+    ]
+    if (visitor?.messagePlatform == null) {
+      promiseArray.push(
+        this.visitorService.update(visitor.id, {
+          messagePlatform: input.value ?? undefined
+        })
+      )
     }
 
-    return await this.eventService.save({
-      ...input,
-      __typename: 'ChatOpenEvent',
-      visitorId: visitor.id,
-      createdAt: new Date().toISOString(),
-      journeyId
-    })
+    const [ChatOpenEvent] = await Promise.all([...promiseArray])
```

Comment:
```suggestion
    const [chatOpenEvent] = await Promise.all([...promiseArray])
```

change to lower case first letter for variable names